### DISCUSSION TITLE
[performance] Is it possible to avoid/delay any actions as much as possible until a PR/MR is raised?

### DISCUSSION THREAD

**Discussion by herndlm** [renovatebot/renovate#21246] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.25.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.6.7-ee

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Save time / improve performance as much as possible by delaying / deferring actions like lockfile updates.

This might be a stupid question, but I was just wondering if it is possible to avoid e.g. lockfile updates of branches where no PR is created because they are still pending. Wouldn't it be enough to just do the lookup / extraction / .. and stop then?

I think in our cases there are some long-pending updates that renovate updates with every run and this feels like a waste of resources and slowly sums up if the number of repositories is high.

### What you tried so far.

decrease `branchConcurrentLimit`

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by herndlm** [comment]
or is this because I have `prCreation` set to `not-pending` and there could be other status checks except pending hours / stability days? 🤔
the config looks like the following
```
...
  "prCreation": "not-pending",
  "prNotPendingHours": 1,
  "rangeStrategy": "update-lockfile",
  "rollbackPrs": true,
  "stabilityDays": 3,
...
```

**Reply by rarkins** [reply]
It will be in pending state if either:
- Other checks exist, which are still pending or
- You have no other checks other than stability days

**Comment by herndlm** [comment]
oh and I have one more, sorry :)

If internal checks such as stabilityDays, do not count as being "green" anymore - does that mean that with the previously mentioned config no PRs are going to be created if no further checks exist?

**Reply by rarkins** [reply]
That's right, although I think we have a timeout which might kick in

**Comment by rarkins** [comment, marked as answer]
Answering your original question, no this is definitely not possible and we don't do it
